convention.md

```diff
@@ -204,6 +211,7 @@ Each property must have a unique property ID on a per-node basis which adhere to
 
 * Properties can be **retained**.
   A property is retained by default. A non-retained property would be useful for momentary events (door bell pressed).
+  Non-retained properties must be sent with a QoS of **Exactly once (2)** (this applies for both the value and the `/set`-command, but not for any of the attributes).
```

Comment:
Do we really want that ?
QOS 2 is something thats really hard to support on devices without persistent storage.
Shouldn't QOS 1 be enough ?

Reply:
See #166 for the discussion about this. I don't know really much about the MQTT implementation internals and their implications.

Would it be a solution to say that non-retained properties ***should*** be sent with QoS 2? So all devices that are capable will do it and the few that can't don't have to.

Reply:
I get why the set command should be QoS 2 in some cases, but why the value ?
Controllers should be able to handle the same value arriving twice just fine.
The problem with mixing QoS values is that mqtt does not guarantee ordering for different QoS values, only for the same QoS value.
See [here](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718105).
So if we mix QoS 1 and 2 for values sent by the device the $state ready message might arrive before the value of a property.

Reply:
@Thalhammer thank you that link explaining message ordering. My prior understanding was there no guarantee on ordering.

The use case for a non retained properties is not clear to me. 

For instance: door bell button, a device to controller event. This is a time sensitive event. If the controller/broker is offline, then there is really no need for the controller to get that message after more than a minute or so. 

Or

For instance: a single command (brew coffee), a controller to device event. This is a time sensitive command. If the coffee maker doesn't get the command within a minute or so then it should probably never get the command.

Reply:
I've removed that sentence since the topic is still subject to discussions.

Reply:
didn't want to revive an old thread, so started a new issue wrt this discussion: https://github.com/homieiot/convention/issues/247